src/azure-cli-core/azure/cli/core/extension/__init__.py

```diff
@@ -305,6 +305,8 @@ def get_extension_modname(ext_name=None, ext_dir=None):
     pos_mods = [n for n in os.listdir(ext_dir)
                 if n.startswith(EXTENSIONS_MOD_PREFIX) and os.path.isdir(os.path.join(ext_dir, n))]
     if len(pos_mods) != 1:
+        logger.warning("Find incomplete extension in '%s'. Please delete this folder and install the extension again.",
+                       ext_dir)
         raise AssertionError("Expected 1 module to load starting with "
                              "'{}': got {}".format(EXTENSIONS_MOD_PREFIX, pos_mods))
```

Comment:
Now that we are already raising error, why not include the message in the `AssertionError`.

Reply:
Thank you, I move the message to AssertError